### ISSUE TITLE
Allow building numcodecs against system blosc

### ISSUE THREAD

**Issue by joshmoore** [zarr-developers/numcodecs#254]
Hi, @joshmoore packaging of numcodecs is already quite straightforward.
Currently there are 4 patches applied (see https://salsa.debian.org/science-team/numcodecs/-/tree/master/debian/patches) but I'm going to remove the last one ASAP in consideration of what discussed in this issue.

Probably the only think that could be improved upstream form a packaging point of view is to provide also the possibility to built numcodecs using system version of the blosc library (instead of using the bundled sources).
This is currently addressed by https://salsa.debian.org/science-team/numcodecs/-/blob/master/debian/patches/0003-No-embedded-libs.patch.

_Originally posted by @avalentino in https://github.com/zarr-developers/numcodecs/pull/249#issuecomment-693198279_

**Comment by jakirkham** [comment]
Agreed. Just noting this would be of interest on the conda-forge side as well.

That said, if there were Blosc wheels, we could go ahead and use Blosc from Python and not need to build this part at all.

**Comment by QuLogic** [comment]
I currently do [this in Fedora](https://github.com/QuLogic/numcodecs/compare/v0.8.0..fedora-patches), but could turn those into a proper PR if there's interest upstream.

**Comment by jakirkham** [comment]
Well at this point Blosc has its own wheels, Conda packages, etc. So ideally we would drop building Bloscs here altogether. That said, it is not quite that straightforward since we are reusing vendored copies of Zstd, LZ4, etc. So these would need to be replaced by dependencies on other packages

**Comment by QuLogic** [comment]
Which is what's done there, unless you mean other _Python_ packages, which is not what's asked here by OP specifically.
